src/librustc_resolve/macros.rs

```diff
@@ -572,7 +572,7 @@ impl<'a> Resolver<'a> {
             ScopeSet::Module => (TypeNS, None, false, false),
         };
         let mut where_to_resolve = match ns {
-            _ if is_absolute_path || is_import && rust_2015 => WhereToResolve::CrateRoot,
+            _ if is_absolute_path => WhereToResolve::CrateRoot,
```

Comment:
When everything is done, this PR will have to wait until https://github.com/rust-lang/rust/pull/58349 lands.